Lesson 2/app/src/main/java/com/mobileapps/lesson2/RegisterFragment.kt

```diff
@@ -0,0 +1,74 @@
+package com.mobileapps.lesson2
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import com.google.android.material.button.MaterialButton
+import com.google.android.material.textfield.TextInputLayout
+
+class RegisterFragment :Fragment() {
+    private lateinit var emailTextInputLayout: TextInputLayout
+    private lateinit var passwordTextInputLayout: TextInputLayout
+    private lateinit var registerButton: MaterialButton
+    private lateinit var loginTextView: TextView
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_register, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        emailTextInputLayout = view.findViewById(R.id.emailTextInputLayout)
+        passwordTextInputLayout = view.findViewById(R.id.passwordTextInputLayout)
+        registerButton = view.findViewById(R.id.nextButton)
+        loginTextView = view.findViewById(R.id.loginTextView)
+
+        loginTextView.setOnClickListener {
+            (activity as? AccountActivity)?.replaceFragment(LoginFragment())
```

Comment:
Casting activity to a particular activity class

Reply:
Also and again - don't decide about navigation from within a fragment. As much as starting another activity from a fragment can be ok, starting one fragment from another is a no-go for me. Explanation was in class - but shortly speaking activities must implement interfaces of fragments running inside of them. If you don't navigate in the activity itself, you will need to check every fragment's implementation to find out which interfaces you should implement in activity